### PR TITLE
squash loki.source.kubernetes http_client_config

### LINKED ISSUE #2995: Flow: squash `http_client_config` in `loki.source.kubernetes`
Given the `http_client_config` blocks in other components are being squashed, it may make sense to do the same for `loki.source.kubernetes`.

### LINKED ISSUE #2996: Flow: squash `http_client_config` in `loki.source.podlogs`
Given the `http_client_config` blocks in other components are being squashed, it may make sense to do the same for `loki.source.podlogs`.

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Description

squash loki.source.kubernetes http_client_config

#### Which issue(s) this PR fixes

Closes #2995 
Closes #2996

#### Notes to the Reviewer

No other places referencing this component's http_client_config were found.

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/loki.source.kubernetes.md [hunk 1]
```diff
@@ -67,14 +67,21 @@ The following blocks are supported inside the definition of
...
+client > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
 
 The `>` symbol indicates deeper levels of nesting. For example, `client >
-http_client_config` refers to an `http_client_config` block defined
+basic_auth` refers to a `basic_auth` block defined
 inside a `client` block.
 
 [client]: #client-block
-[http_client_config]: #http_client_config-block
+[basic_auth]: #basic_auth-block
```
Comment (erikbaranowski) [inline docs/sources/flow/reference/components/loki.source.kubernetes.md:81]: these were missing from the file

File: docs/sources/flow/reference/components/loki.source.kubernetes.md [hunk 2]
```diff
@@ -89,13 +96,34 @@ Name | Type | Description | Default | Required
...
+ At most one of the following can be provided:
+ - [`bearer_token` argument][client].
+ - [`bearer_token_file` argument][client]. 
+ - [`basic_auth` block][basic_auth].
+ - [`authorization` block][authorization].
+ - [`oauth2` block][oauth2].
 
-The `http_client_config` block configures settings used to connect to the
-Kubernetes API server.
+### basic_auth block
```
Comment (erikbaranowski) [inline docs/sources/flow/reference/components/loki.source.kubernetes.md:112]: these were missing from the file

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ Main (unreleased)
 
 - Some blocks in Flow components have been merged with their parent block to make the block hierarchy smaller:
   - `prometheus.scrape > client > http_client_config` is merged into the `client` block. (@erikbaranowski)
+  - `loki.source.kubernetes > client > client> http_client_config` is merged into the `client > client` block. (@erikbaranowski)
```
Comment (rfratto) [inline CHANGELOG.md:21]: Are you sure this is right? There's two levels of client blocks?

Comment (erikbaranowski) [reply CHANGELOG.md:21]: I've taken the first > client to represent the top level in these comments so there is another layer on this component with exactly that name.

**Review by rfratto — APPROVED** [review]

**Pull request merged**